kolibri/core/device/api.py

```diff
@@ -24,3 +33,46 @@ def create(self, request, *args, **kwargs):
         data = serializer.save()
         output_serializer = self.get_serializer(data)
         return Response(output_serializer.data, status=status.HTTP_201_CREATED)
+
+
+class DeviceInfoView(views.APIView):
+
+    permission_classes = (UserHasAnyDevicePermissions, )
+
+    def get(self, request, format=None):
+        info = {}
+
+        info['version'] = kolibri.__version__
+
+        status, urls = get_urls()
+        if not urls:
+            # Will not return anything when running the debug server, so at least return the current URL
+            urls = [request.build_absolute_uri('/')]
+
+        filtered_urls = [url for url in urls if '127.0.0.1' not in url and 'localhost' not in url]
+
+        if filtered_urls:
+            urls = filtered_urls
+
+        info['urls'] = urls
+
+        if settings.DATABASES['default']['ENGINE'].endswith('sqlite3'):
+            # If any other database backend, will not be file backed, so no database path to return
+            info['database_path'] = settings.DATABASES['default']['NAME']
+
+        info['device_name'] = platform.node()
+        info['device_id'] = InstanceIDModel.get_or_create_current_instance()[0].id
+        info['os'] = platform.platform()
+
+        # Will either return a named tuple or None if the drive cannot be found
+        drive = get_drive_data(get_content_storage_folder_path())
+
+        if drive:
+            info['content_storage_free_space'] = drive.freespace
```

Comment:
Can update this when #2482 has been merged, which has a utility specifically for this!